### PR TITLE
MNT Add PHP 8 build

### PR DESCRIPTION
**Pull request by Cheddam**
As part of https://github.com/silverstripe/silverstripe-framework/issues/9704

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,7 +20,7 @@
         "silverstripe/versioned": "1.x-dev"
     },
     "require-dev": {
-        "phpunit/phpunit": "^5.7",
+        "sminnee/phpunit": "^5.7",
```
[the remark this answers is not in the document]
Comment (robbieaverill) [reply composer.json:23]: We've had to fork it for compatibility with the wide range of PHP versions we support - for more info see https://github.com/silverstripe/silverstripe-framework/issues/9704

[the remark this answers is not in the document]
Comment (Firesphere) [reply composer.json:23]: Ah, okay.

Out of the spirit of "Share the love - Open source", are the changes something that could be merged upstream to PHPUnit?

**Comment by Cheddam** [comment]
The remaining CI failure is a PostgreSQL issue also present on the `4` branch.

**Review by dnsl48 — APPROVED** [review]
good!

**Pull request merged**